test/index.js

```diff
@@ -112,6 +112,17 @@ describe('Raw Body', function () {
     })
   })
 
+  it('should throw if stream is not readable', function (done) {
+    var stream = new Readable()
+    stream.readable = false
```

Comment:
Can this be made more realistic? For example, what makes a readable stream unreadable in the situation you encountered?

Reply:
the stream is a instance of http.IncomingMessage in my project, which will be unreadable after `end` event emitted. but I have no idea how to create it manually in unit test, these code maybe helpful.
```javascript
const koa = require("koa");
const raw = require("raw-body");

const app = new koa();

app.use(async(ctx) => {
    ctx.request.body = await raw(ctx.req);
    console.log('ctx will be unreadable :: ', ctx.req.readable);

    // request will hang there without readable check
    // ctx.request.body = await raw(ctx.req);

    ctx.body = 'foo';
});

app.listen(3000);
```

Reply:
We unit test http instance streams in this file: https://github.com/stream-utils/raw-body/blob/master/test/http.js which may help you see how to do it 👍

Reply:
ok, I got it.